lib/promscrape/discoveryutils/client.go

```diff
@@ -247,8 +247,14 @@ func (c *Client) getAPIResponseWithParamsAndClientCtx(ctx context.Context, clien
 		return nil, fmt.Errorf("cannot create request for %q: %w", requestURL, err)
 	}
 
-	c.setHTTPHeaders(req)
-	c.setHTTPProxyHeaders(req)
+	err = c.setHTTPHeaders(req)
+	if err != nil {
+		return nil, fmt.Errorf("cannot create request for %q: %w", requestURL, err)
```

Comment:
error should mention the exact action that failed. In this case it should mention that it failed to set headers.

Reply:
Fixed!